app/src/main/java/com/aguesoftguar/medalarm/data/Medicine.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2016, Agüesoftgüar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aguesoftguar.medalarm.data;
+
+import com.google.common.base.Objects;
+import com.google.firebase.database.Exclude;
+
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Immutable model class for a medicine.
+ */
+public class Medicine {
+
+   // Required parameters
+   private String name;
+
+   private String patientUuid;
+
+   private String initialDateDay;
+
+   private String initialDateHour;
+
+   private String dosesInterval;
+
+   @Nullable
+   private String finalDateDay;
+
+   @Nullable
+   private String finalDateHour;
+
+   private int reminder;
+
+   private int color;
+
+   private boolean photo;
+
+   /**
+    * We have to use the Builder pattern.
+    * Instead of making the desired object directly, the client calls
+    * a constructor (or static factory) with all of the required parameters
+    * and gets a builder object. Then the client calls setter-like methods on
+    * the builder object to set each optional parameter of interest.
+    * Finally, the client calls a parameterless build method to generate the object,
+    * which is immutable. The builder is a static member class of the class it builds.
+    */
+   public static class MedicineBuilder {
+      // Required parameters
+      private String name;
+
+      private String patientUuid;
+
+      private String initialDateDay;
+
+      private String initialDateHour;
+
+      private String dosesInterval;
+
+      // Optional parameters - initialized to default values
+      @Nullable
+      private String finalDateDay = "";
```

Comment:
I think `@Nullable` is not needed if you initialize the value using "" as default value. If the value can really be `null`, maybe you could do the following: 
```
 @Nullable
 private String finalDateDay;
```